Create dummy vodacom endpoints

beesaferoot opened the pull request:
<!-- First of all, thank you for your contribution to this repository! -->

<!-- Please give the Pull Request a meaningful title for the release notes -->

### Brief summary of the change made

<!-- Please include `closes: #XXXX` to automatically close any corresponding issue when the pull request is merged. Alternatively if not fully closed you can say `makes progress on: #XXXX`.-->

This PR adds dummy endpoints as described in #489 
### Are there any other side effects of this change that we should be aware of?

### Describe how you tested your changes?

<!-- For manual testing-please provide detailed steps, screenshots, etc.. -->
<!-- If the repository provides unit tests, please add test cases covering the changes. -->

### Pull Request checklist

Please confirm you have completed any of the necessary steps below.

- [x] Meaningful Pull Request title and description
- [ ] Changes tested as described above
- [ ] Added appropriate documentation for the change.
- [ ] Created GitHub issues for any relevant followup/future enhancements if appropriate.

dmohns commented on src/backend/app/Http/Controllers/VodacomTransactionController.php:
```diff
@@ -2,10 +2,18 @@
 use Illuminate\Http\Request;
 
 class VodacomTransactionController extends Controller {
+    private VodacomService $vodacomService;
+
+    public function __construct(VodacomService $vodacomService) {
+        $this->vodacomService = $vodacomService;
+    }
```
You can use (so-called) constructor property promotion here to simplify this to

```suggestion
    public function __construct(
        private VodacomService $vodacomService
    ) {}
```

beesaferoot replied:
Updated.

dmohns commented:
One more general comment: For sharing this with Vodacom it would be important that this new endpoint shows up in our OpenAPI definitions. 

Could you ...

1. Check that `scribe` picks it up correctly? The process is semi-manual and documented here: https://github.com/EnAccess/micropowermanager/blob/main/docs/README.md#generate-openapi-docs
2. Add the updated OpenAPI to this PR as well?
3. Check if we have provided enough meta descriptions that the OpenAPI docs are meaningful to an external developer?

beesaferoot commented:
> One more general comment: For sharing this with Vodacom it would be important that this new endpoint shows up in our OpenAPI definitions.
> 
> Could you ...
> 
> 1. Check that `scribe` picks it up correctly? The process is semi-manual and documented here: https://github.com/EnAccess/micropowermanager/blob/main/docs/README.md#generate-openapi-docs
> 2. Add the updated OpenAPI to this PR as well?
> 3. Check if we have provided enough meta descriptions that the OpenAPI docs are meaningful to an external developer?

Will look into this.

beesaferoot commented:
> One more general comment: For sharing this with Vodacom it would be important that this new endpoint shows up in our OpenAPI definitions.
> 
> Could you ...
> 
> 1. Check that `scribe` picks it up correctly? The process is semi-manual and documented here: https://github.com/EnAccess/micropowermanager/blob/main/docs/README.md#generate-openapi-docs
> 2. Add the updated OpenAPI to this PR as well?
> 3. Check if we have provided enough meta descriptions that the OpenAPI docs are meaningful to an external developer?


Updated the OpenAPI spec, added endpoints descriptions. Please review  

@dmohns

beesaferoot commented:
> Nice, two comments on the OpenAPI
> 
> 1. Somehow the amount<>number data type is not correctly picked up. It says `any` and the example payload is also misleading. Could you look into that?
> 
> <img alt="image" width="877" src="https://private-user-images.githubusercontent.com/14202480/415570380-80c37727-4456-4e36-9ed3-107a28538c35.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAxMjg1ODYsIm5iZiI6MTc0MDEyODI4NiwicGF0aCI6Ii8xNDIwMjQ4MC80MTU1NzAzODAtODBjMzc3MjctNDQ1Ni00ZTM2LTllZDMtMTA3YTI4NTM4YzM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjIxVDA4NTgwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmNzRmZDM1OTNkZGIzMGViZmVkNzAyZTgwNzk2ZWRmMjdiNzM5Zjc5ZGE3ZmI4MDIzNjdkOWY2YjA5MThiYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1LO9qXMY_90a708Xr9gfMAq47SkOhDd2Gy0HktgIGiY">
> 2. For the `payerPhoneNumber`. I would love to have a more precise example/documentation what is our understanding of "international format". Could you maybe look into how we can use a `pattern` here: https://swagger.io/docs/specification/v3_0/data-models/data-types/#strings

The examples seems auto generated by scribe; there should be a way to get it to work correctly.

>  2. For the `payerPhoneNumber`. I would love to have a more precise example/documentation what is our understanding of "international format". Could you maybe look into how we can use a `pattern` here: https://swagger.io/docs/specification/v3_0/data-models/data-types/#strings

I will do just. 


For the issue #497, is it possible we can include more details to the response and request? This will help make the documentation easier and perhaps better aligned.
